napari/components/overlays/base.py

```diff
@@ -0,0 +1,20 @@
+from ...utils.events import EventedModel
+from .._viewer_constants import CanvasPosition
+
+
+class Overlay(EventedModel):
```

Comment:
We've got quite a lot of classes here - not sure if this is a pattern @jni would want us to go too right away. Maybe we just start with `Overlay` with everything on it and drop CanvasOverlay / SceneOverlay

Reply:
Ah ok I just saw in #4894

> CanvasOverlay: things that live in canvas space (ScaleBar, TextOverlay, InteractionBox, the welcome screen, and in the future maybe ColorBars and other HUD-like visuals) and thus should be children of canvas.view
> SceneOverlay: things that live in scene space -- also referred to as "displayed coordinates"; this is the 2D or 3D slices being rendered -- and thus should be children of the canvas.view.scene

So I see how these are two different types of thing - still not quite clear to me they need their own classes, but maybe. If so it would be good to have some nice doc strings for the them, and maybe make `_Overlay` or `OverlayBase` so that it was clear it was not something to be used directly

Reply:
It's definitely reasonable to make them private for now.

The reason for having different classes is mainly because in #4894 I used `isinstance` checks to choose how to parent the visual. This can probably be done with an attribute just as well, so I'm not particularly attached to class separation; I just found it more intuitive.